python/tests/test_drawing.py

```diff
@@ -1435,34 +1440,35 @@ def verify_basic_svg(self, svg, width=200, height=200):
             cls = group.attrib["class"]
             assert re.search(r"\broot\b", cls)
 
-    def test_draw_file(self):
+    def test_draw_file(self, tmp_path):
+        # NB - deleting tmp_path dir unnecessary: this is done after 3 times by pytest
         t = self.get_binary_tree()
-        fd, filename = tempfile.mkstemp(prefix="tskit_viz_")
-        try:
-            os.close(fd)
-            svg = t.draw(path=filename)
-            assert os.path.getsize(filename) > 0
-            with open(filename) as tmp:
-                other_svg = tmp.read()
-            assert svg == other_svg
-            os.unlink(filename)
-
-            svg = t.draw_svg(path=filename)
-            assert os.path.getsize(filename) > 0
-            with open(filename) as tmp:
-                other_svg = tmp.read()
-            self.verify_basic_svg(svg)
-            self.verify_basic_svg(other_svg)
-
-            ts = t.tree_sequence
-            svg = ts.draw_svg(path=filename)
-            assert os.path.getsize(filename) > 0
-            with open(filename) as tmp:
-                other_svg = tmp.read()
-            self.verify_basic_svg(svg)
-            self.verify_basic_svg(other_svg)
-        finally:
-            os.unlink(filename)
+        filename = tmp_path / "tree-draw.svg"
+        print(f"Saving tree.draw() to {filename}")  # `pytest -rP` shows path to SVG
```

Comment:
I'm not keen on putting in prints. Won't this spam the output everytime someone runs ``pytest -vs``?

Reply:
>Won't this spam the output everytime someone runs `pytest -vs`?

It doesn't seem to, no (try it). I think stdout is normally captured by something else. The `-rP` is required to get anything out (it took me a while to find that). It's pretty useful being able to look at the files that have been created by the test suite, as they are in very obscure directories (e.g. `/private/var/folders/00/l36df9q14hv13hhw8wglncch0000gr/T/pytest-of-yan/pytest-541/popen-gw3/test_draw_file0/ts-draw_svg.svg`) so finding them is hard.

Reply:
It does. Run with ``-n0 -vs``, and it prints all outputs as they are generated. I use this all the time.

``-rP`` just means "show the output" from passed tests (which is only useful to you now because I've nagged about leaving stray prints in innumerable reviews).

The location is pretty logical for Linux users ( /tmp/pytest-of-jk/pytest-13/test_draw_file0/), and you can override the default location if you want with [--basetmpdir](https://docs.pytest.org/en/stable/tmpdir.html#the-default-base-temporary-directory).

So yes, please delete the prints.

Reply:
Ah, it requires the `-n0` to print, I see.

I can delete. I would like to leave a comment about --basetmpdir somewhere though, as that'll be super useful for testing, and I'm bound to forget. Perhaps in a comment at the top of that test.

Reply:
Sure - put in whatever comments you like (keeping the prints commented out, seems like something that'll handy in future, too).